HarClient/HarMessageHandler.cs

```diff
@@ -54,13 +84,53 @@ public HarMessageHandler(HttpMessageHandler innerHandler) : base(innerHandler)
             DateFormatHandling = DateFormatHandling.IsoDateFormat,
         };
 
-        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        /// <summary>
+        /// Gets or sets the effective redirect policy, which determines whether redirect messages
+        /// are recorded in the HAR, and whether they are produced directly
+        /// </summary>
+        public RedirectPolicy RedirectPolicy
         {
-            var entrySource = new HarEntrySource(request, DateTime.Now);
-            entries.Add(entrySource);
-            var response = await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
-            entrySource.SetResponse(response);
-            return response;
+            get => redirectPolicyImplementation.Policy;
+            set => redirectPolicyImplementation.Policy = value;
+        }
+
+        protected override async Task<HttpResponseMessage?> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var nextRequest = request ?? throw new ArgumentNullException(nameof(request));
+            var redirectSet = new HashSet<Uri>
+            {
+                nextRequest.RequestUri
+            };
+            while (true)
+            {
+                var response = await RequestOne(nextRequest, cancellationToken).ConfigureAwait(false);
+                if (response != null &&
+                    response.IsRedirect(out var location) &&
```

Comment:
Most likely we should not e.g. redirect a POST; at least not with another POST.

Research required.

Reply:
Tested native behaviour, and reimplemented that.